packages/calcite-components/src/components/stepper/stepper.tsx

```diff
@@ -338,6 +383,48 @@ export class Stepper implements LocalizedComponent, T9nComponent {
     });
   }
 
+  private determineActiveStepper(currentPositionChanged = false): void {
+    const totalItems = this.items.length;
+    if (!this.elWidth || !totalItems || this.layout !== "horizontal" || totalItems === 1) {
+      return;
+    }
+
+    const activePosition = this.currentPosition || 0;
+    const totalMinWidthOfItems = totalItems * this.getMinWidthOfStepperItem();
+    const totalRowGap = (totalItems - 1) * (parseInt(window.getComputedStyle(this.el).rowGap) || 0);
+
+    if (this.elWidth <= totalMinWidthOfItems + totalRowGap) {
+      if (this.singleViewMode && !currentPositionChanged) {
+        return;
+      }
+      this.el.style.gridTemplateColumns = "none";
+      this.singleViewMode = true;
+      this.el.style.display = "flex";
+      this.el.style.flexDirection = "columns";
+
+      this.items.forEach((item: HTMLCalciteStepperItemElement, index) => {
+        if (index !== activePosition) {
+          item.style.display = "none";
+        } else {
+          item.style.display = "contents";
+          item.singleViewMode = true;
+        }
+      });
+    } else if (this.elWidth > totalMinWidthOfItems + totalRowGap) {
+      if (!this.singleViewMode) {
+        return;
+      }
+      this.el.style.display = "grid";
```

Comment:
Can you look at pushing static styles to a CSS class and only use inline styles for computed values?